WooCommerce/Classes/ViewModels/MainTabViewModel.swift

```diff
@@ -0,0 +1,77 @@
+import Foundation
+import Yosemite
+
+
+// MARK: - MainTabViewModel Notifications
+//
+extension NSNotification.Name {
+
+    /// Posted whenever an OrderBadge refresh is required.
+    ///
+    public static let ordersBadgeReloadRequired = Foundation.Notification.Name(rawValue: "com.woocommerce.ios.ordersBadgeReloadRequired")
+}
+
+final class MainTabViewModel {
+    /// Callback to be executed when this view model receives new data
+    /// passing the string to be presented in the badge as a parameter
+    ///
+    var onReload: ((String?) -> Void)?
```

Comment:
Since this view model is for the whole tab bar, not an individual tab, do you think this closure could be applicable to all tabs like `onReload: ((String?, WooTab) -> Voice)?`

Reply:
Good point, this is abstract (onReload) and yet specific (it only works for the orders badge! 

I am thinking that maybe it'd be even better to mark this method clearly as `onReloadBadge`, and if/when we need more reasons to reload the tab bar, due to more concerns, we could add methods specific to those concerns. 

That way we'd avoid a call site with code like this:

```
        viewModel.onReload = { [weak self] aString, aTab in
            guard let self = self else {
                return
            }

            if aTab == .first {
                // do A
            }

            if aTab == .second {
                // do B
            }
        }
```

Also, we'd be contrained by the type off the value we actually pass as a parameter to the closure (in this case. it's a string)